lib/io/hardware_can.cpp

```diff
@@ -109,10 +109,12 @@ core::Result HardwareCan::receive()
     logger_.log(core::LogLevel::kFatal, "No CanProccessor associated with id %i", message.can_id);
     return core::Result::kFailure;
   }
+  core::Result process_successful = core::Result::kSuccess;
```

Comment:
Why do you make this object instead of just returning `kFailure` when the condition is met and `kSuccess` otherwise?